crates/jetstreamer-source/src/lib.rs

```diff
@@ -0,0 +1,565 @@
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use futures_util::FutureExt;
+use jetstreamer_firehose::firehose::{firehose, BlockData, TransactionData};
+use tokio::sync::mpsc::Sender;
+use tokio_util::sync::CancellationToken;
+use tracing::{debug, error, info};
+use yellowstone_grpc_proto::{
+    geyser::{subscribe_update::UpdateOneof, SubscribeUpdate, SubscribeUpdateBlock},
+    solana::storage::confirmed_block::{BlockHeight, UnixTimestamp},
+};
+use yellowstone_vixen::{sources::SourceTrait, Error as VixenError};
+use yellowstone_vixen_core::Filters;
+
+struct VixenStreamHandler {
+    tx: Sender<Result<SubscribeUpdate, yellowstone_grpc_proto::tonic::Status>>,
+    filters: Filters,
+    config: JetstreamSourceConfig,
+}
+
+impl VixenStreamHandler {
+    fn new(
+        tx: Sender<Result<SubscribeUpdate, yellowstone_grpc_proto::tonic::Status>>,
+        filters: Filters,
+        config: JetstreamSourceConfig,
+    ) -> Self {
+        Self {
+            tx,
+            filters,
+            config,
+        }
+    }
+
+    async fn process_block(&self, block: BlockData) -> Result<(), Error> {
+        info!(slot = block.slot(), "Processing block");
+
+        match block {
+            BlockData::Block {
+                parent_slot,
+                parent_blockhash,
+                slot,
+                blockhash,
+                rewards,
+                block_time,
+                block_height,
+                executed_transaction_count,
+                entry_count,
+            } => {
+                let update = SubscribeUpdate {
+            filters: vec![],
```

Comment:
@senzenn Vixen uses this filter list to match a subscribe update with a parser so you need to include the filter key for Vixen to match on when it goes through the runtime.